internal/nfs/controller/volume.go

```diff
@@ -251,18 +253,18 @@ func (nv *NFSVolume) deleteExportCommand(cmd, nfsCluster string) []string {
 // getNFSCluster fetches the NFS-cluster name from the CephFS journal.
 func (nv *NFSVolume) getNFSCluster() (string, error) {
 	if !nv.connected {
-		return "", fmt.Errorf("can not get NFS-cluster for %q: not connected", nv)
+		return "", fmt.Errorf("can not get NFS-cluster for %q: %w", nv, ErrNotConnected)
 	}
 
 	fs := fscore.NewFileSystem(nv.conn)
 	fsName, err := fs.GetFsName(nv.ctx, nv.fscID)
 	if err != nil {
-		return "", fmt.Errorf("failed to get filesystem name for ID %x: %w", nv.fscID, err)
+		return "", fmt.Errorf("filesystem name for ID %x %w: %v", nv.fscID, ErrNotFound, err)
```

Comment:
What if there is some other error while trying to fetch fsname or metadatapool. 
This change assumes, any error to be ErrNotFound and continues deletion. 
There maybe chances of intermittent connection errors or something else.

Reply:
Yes, but I do not think that these errors can be checked easily. Do you prefer to handle the error for the filesystem-id as a generic error?

Reply:
> Yes, but I do not think that these errors can be checked easily. Do you prefer to handle the error for the filesystem-id as a generic error?

Any error other than fs/metadatapool not found can lead to stale nfs exports in this case, right ? 
Maybe check for 'not found' in the actual error msg, only then add this `ErrNotFound` so it can be checked in higher levels ?

Reply:
fs.GetFsName already returning util.ErrPoolNotFound error if the fs doesn't exist just use it?